static/js/stream_sort.js

```diff
@@ -26,7 +26,14 @@ function compare_function(a, b) {
 
 export function sort_groups(streams, search_term) {
     const stream_id_to_name = (stream) => sub_store.get(stream).name;
-    streams = util.filter_by_word_prefix_match(streams, search_term, stream_id_to_name);
+    // Use -, _ and / as word separators apart from the default space character
+    const word_separator_regex = /[\s/_-]/;
```

Comment:
Moved the `-` to the end of the bracketed region, to make clear we're not using `-` as a range here. I didn't test sure if your original version was buggy in considering every character in the the range between two of the other 3 characters here as a separator, but I think it might have been.